src/libmongoc/doc/mongoc_client_set_timeout_ms.rst

```diff
@@ -0,0 +1,32 @@
+:man_page: mongoc_client_set_timeout_ms
+
+mongoc_client_set_timeout_ms()
+==============================
+
+Synopsis
+--------
+
+.. code-block:: c
+
+  bool
+  mongoc_client_set_timeout_ms (mongoc_client_t *client, int64_t *timeout_ms, bson_error_t *error)
+
+Sets the client-side operation timeout value for ``client`` to the given ``timeout_ms``. If there is an error setting the timeout, this method will return false and ``error`` will be set. If ``client`` belongs to a :symbol:`mongoc_client_pool_t`, its timeout will be reset to match the pool's timeout once it is pushed back into the pool.
```

Comment:
This last sentence makes me wonder about other modifications made to clients checked out from the pool.

Suppose a pool is created with `w:majority` via its URI. And suppose a client is popped, modified with `w:1` (or worse), and later pushed back into the pool. When some other thread happens to pop the same client from the pool, would it be in for a surprise if it expected the client to have the original WC from the URI (`w:majority`)? If so, this sounds like a general bug worthy of a ticket (e.g. "Revert state `mongoc_client_t` when pushing back to pool").

Reply:
Yes! I filed https://jira.mongodb.org/browse/CDRIVER-3537 some time ago when I noticed this for read concern.

Reply:
this sent me into a panic thinking we might be relying on the "reset" assumption in Swift but fortunately, we are not 🙂

Reply:
You're right, that does sound like potential surprising and buggy behavior!  We currently don't make any alterations to the client when we return it to the pool, so it could be "dirty" in unexpected ways.  I've opened https://jira.mongodb.org/browse/CDRIVER-3878 to investigate that.

Reply:
Ah whoops closed my ticket as a duplicate :D

Reply:
Resolving, since [CDRIVER-3537](https://jira.mongodb.org/browse/CDRIVER-3537) exists.